pkg/hs/urbit-king/lib/Urbit/Vere/Ames.hs

```diff
@@ -106,13 +114,31 @@ ames'
   -> (Text -> RIO e ())
   -> RIO e ([Ev], RAcquire e (DriverApi NewtEf))
 ames' who isFake stderr = do
+  -- Unfortunately, we cannot use TBQueue because the only behavior
+  -- provided for when full is to block the writer, and we want to
+  -- instead drop the incoming packet on the floor. The implementation
+  -- below uses materially the same data structures as TBQueue, however.
   ventQ :: TQueue EvErr <- newTQueueIO
+  avail :: TVar Word <- newTVarIO queueBound
+  let
```

Comment:
This is super nitty, but most of the king code uses `where` instead of `let` for inline function definitions. I get that this is here so you don't have to manually pass `avail` or `ventQ` to `enqueuePacket` or `dequeuePacket`, so maybe ignore me?

Reply:
So this is let *in a do block*, which is actually a totally different thing; it compiles down to fmap or something I think.